core/metrics/src/pusher.rs

```diff
@@ -54,12 +54,36 @@ pub async fn push_forever(host_url: &str) {
 }
 
 pub async fn push(client: &Client, push_url: String) {
-    let current_metrics = crate::service::export_metrics().await;
+    let metrics = crate::service::export_metrics().await;
     let res = client
         .put(push_url.as_str())
-        .send_body(current_metrics)
+        .send_body(metrics.clone())
         .await;
-    log::trace!("Pushed current metrics {:#?}", res);
+    match res {
+        Ok(r) if r.status().is_success() => {
+            log::trace!("Metrics pushed: {}", r.status())
+        }
+        Ok(r) if r.status().is_server_error() => {
+            log::debug!("Metrics server error: {:#?}", r);
+            log::trace!("Url: {}\nMetrics:\n{}", push_url, metrics);
+        }
+        Ok(mut r) => {
+            let body = r.body().await.unwrap_or_default().to_vec();
+            let msg = String::from_utf8_lossy(&body);
+            log::warn!("Pushing metrics failed: `{}`: {:#?}", msg, r);
```

Comment:
in case of other errors we want to `warn` users, especially developers while they are testing their changes, to not allow brokerage of the metrics reporting again.